tidy3d/plugins/near2far/near2far.py

```diff
@@ -1,130 +1,483 @@
 """Near field to far field transformation plugin
 """
+from typing import List, Dict
 import numpy as np
-
-from ...constants import C_0, ETA_0
-from ...components.data import SimulationData
+import xarray as xr
+import pydantic
+
+from ...constants import C_0, ETA_0, MICROMETER
+from ...components.data import SimulationData, FieldData
+from ...components.monitor import FieldMonitor
+from ...components.types import Direction, Axis, Size
+from ...components.medium import Medium
+from ...components.validators import assert_plane
 from ...log import SetupError
 
-# TODO: implement new version with simulation.discretize
 
+class Near2FarSurface(pydantic.BaseModel):
+    """Data structure to store surface monitor data with associated surface current densities."""
+
+    monitor: FieldMonitor = pydantic.Field(
+        ...,
+        title="Field monitor",
+        description=
+        "Object of :class:`.FieldMonitor` on which near fields will be sampled and integrated.",
+    )
+
+    normal_dir: Direction = pydantic.Field(
+        '+',
+        title="Normal vector orientation",
+        description=
+        ":class:`.Direction` of the surface monitor's normal vector w.r.t.\
+ the positive x, y or z unit vectors. Must be one of '+' or '-'.",
+    )
+
+    size: Size = pydantic.Field(
```

Comment:
This is already covered by `monitor.size` so let's remove.

Reply:
I was forced to do this because of the validator `assert_plane`, which needs to have a `size` field. Is there another way to give this validator access to the monitor's size field?

Reply:
Might be easier to just make a new validator in here

Reply:
ok good point.  Yea let's just Make a new validator on `"monitor"` that errors if `monitor.size.count(0) != 1` and forget about asserts_plane and extra size.  sound good?